### ISSUE TITLE
Analysis branch build throttling not effective (enough)

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#954]
We deployed the analysis branch build throttling on Feb 12 but it clearly isn't effective:

<img width="492" alt="108955279-2e49ca00-766e-11eb-9363-e682090fe222" src="https://user-images.githubusercontent.com/65520/108955565-af08c600-766e-11eb-8693-cede3ecce002.png">

<img width="354" alt="CleanShot 2021-02-24 at 07 03 33@2x" src="https://user-images.githubusercontent.com/65520/108955442-7537bf80-766e-11eb-9983-1a33678d30d0.png">

The initial spike can be explained, as we moved the throttling from the trigger to the analysis stage and lots of throttled builds came online. However, it hasn't caught up as expected.

**Comment by finestructure** [comment]
Branch versions are being suppressed as expected however: Yesterday afternoon's push to LeftPad `master` has been throttled.

<img width="1147" alt="CleanShot 2021-02-24 at 06 23 15@2x" src="https://user-images.githubusercontent.com/65520/108955746-f7c07f00-766e-11eb-8761-4f69f210e847.png">

<img width="1405" alt="CleanShot 2021-02-24 at 06 23 31@2x" src="https://user-images.githubusercontent.com/65520/108955761-fbec9c80-766e-11eb-89b4-3c561e438779.png">

**Comment by finestructure** [comment]
There's a decent amount of throttled revisions, too:

<img width="1157" alt="CleanShot 2021-02-24 at 07 08 33@2x" src="https://user-images.githubusercontent.com/65520/108955903-2e969500-766f-11eb-9b9f-76ac6b9eaaaa.png">

**Comment by finestructure** [comment]
Running the current build trigger selection query vs the throttled query shows that we were throttling more when we were suppressing builds inside the trigger stage:

- current build trigger candidate selection: 87 candidates
- throttled build trigger candidate selection: 48 candidates

So there's a delta of 39 candidates that we were suppressing when we throttled inside TriggerBuilds that we're not suppressing when throttling branch versions inside Analysis.

**Comment by finestructure** [comment]
One difference I just noticed is that in the build trigger throttling we were using `versions.created_at` for the timestamp, whereas in the analysis throttling we're looking at `versions.commit_date`. That changes the candidate count to

- throttled build trigger candidate selection: 52 candidates

Not a huge difference but it makes sense: using `created_at` will skew versions to later times, making fewer versions eligible.

**Comment by finestructure** [comment]
I think I've found an example illustrating the difference in throttling. https://github.com/hummingbird-project/hummingbird.git had two commits to `main` yesterday (~17h and ~19h ago, as of writing):

![CleanShot 2021-02-24 at 11 10 07@2x](https://user-images.githubusercontent.com/65520/108985418-50a10f00-7691-11eb-91fe-410a13d04fcc.png)

We have the `e24ed67` revision currently in our db as latest `defaultBranch`:

<img width="974" alt="CleanShot 2021-02-24 at 11 14 10@2x" src="https://user-images.githubusercontent.com/65520/108985533-73cbbe80-7691-11eb-9402-22f8e2380bf0.png">

The revision was added, because while `e24ed67` is less than 24h old, the next oldest revision before that was from Feb 21, i.e. more than 24h before `e24ed67` - so the new revision went in. The one following, `40c08e8` was correctly suppressed.

Running the current build candidate selection shows the `e24ed67` is a candidate, simply because it doesn't have any builds.

Running the previous throttling candidate selection does _not_ select `e24ed67` - because it is less than 24h old. It should be picked up by the throttled selection in 24h-19h = ~5h (i.e. by ~15:30 UTC / 16:30 CET this afternoon).

**Comment by finestructure** [comment]
I've manually confirmed this scenario for a few other packages that are suppressed by the previous build trigger throttling. All had commits yesterday which were being too aggressively suppressed by the previous logic.

The current algorithm is working as intended as far as I can see and the current load is simply due to demand. The fact that previously we selected fewer candidates is due to an undesirable algorithm.

[Throttling comparison.numbers.zip](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/files/6036226/Throttling.comparison.numbers.zip)
